sdk/nodejs/runtime/resource.ts

```diff
@@ -46,10 +48,11 @@ interface ResourceResolverOperation {
     providerRef: string | undefined;
     // All serialized properties, fully awaited, serialized, and ready to go.
     serializedProps: Record<string, any>;
-    // A set of dependency URNs that this resource is dependent upon (both implicitly and explicitly).
-    dependencies: Set<URN>;
-    // A map from property name to a list of dependency URNs
-    propertyDependencies: Record<string, URN[]>;
+    // A set of URNs that this resource is transitively dependent upon.
+    allDependentResourceURNs: Set<URN>;
+    // Set of URNs that this resource is transitively dependent upon, keyed by the property that
+    // causes the dependency.  All urns in this map must exist in [allDependentResourceURNs]
+    propertyToDependentResourceURNs: Map<string, Set<URN>>;
```

Comment:
note: i renamed these as i was getting totally confused around variables that were pointing at resources, and similarly named variables that were pointing to URNs.